src/parser/jobs/mnk/modules/Forms.tsx

```diff
@@ -21,16 +23,16 @@ export class Forms extends Analyser {
 	@dependency private downtime!: Downtime
 	@dependency private suggestions!: Suggestions
 
-	private forms: number[] = []
-	private opoOpoSkills: number[] = []
+	private forms: Array<Status['id']> = []
+	private opoOpoSkills: Array<Action['id']> = []
 
 	private formless: number = 0
 	private resetForms: number = 0
 	private skippedForms: number = 0
 	private droppedForms: number = 0
 
-	private lastFormChanged?: number
-	private lastFormDropped?: number
+	private lastFormChanged: number | undefined
+	private lastFormDropped: number | undefined
```

Comment:
There's no difference in effective type with these changes.  Personally, I would prefer the ?: format, especially for things that don't exist initially, eventually get set, and then keep that value or get updated to a new value but never back to unset/undefined.

Reply:
It's clearer about the separation between optional and potentially undefined. Due to how Forms work, these vars aren't optional.